### PR TITLE
Fix blank emails when registering

### PR DESCRIPTION
**Pull request by equirke**
Hello,
An attempt at fixing #47.

HTML forms submits fields left empty as blank strings (or at least flask thinks they are blank strings).
Fixing it client side involves adding a JavaScript function to the submit event for the form to delete fields that are blank but I feel you'd rather not do that.

I just included a check for blank strings in the None checks in create_user and update_user. 
Added a check constrain to the user_emails table as requested in #47. Deletes all user_emails where the email is blank as part of adding that constraint

Changed the user-settings templates to use a function that returns '' if the email is None as otherwise jinja renders the text 'None' in the field.

The only place this might have effect (from what I could see) is when interacting with stripe but there is already a check that the user email contains a vaguely email looking string, and just retries without an email if it fails anyway.

Thanks.

### CODE REVIEW

File: csvbase/models.py [hunk 1]
```diff
@@ -81,7 +81,10 @@ class User(Base):
 
 class UserEmail(Base):
     __tablename__ = "user_emails"
-    __table_args__ = (METADATA_SCHEMA_TABLE_ARG,)
+    __table_args__ = (
+        CheckConstraint("email_address != ''", "email_address_not_blank"),
```
Comment (calpaterson) [inline csvbase/models.py:85]: Possibly this is better as `email_address ~ '@'` which checks that whatever is in there is at least a plausible email address.  That's the same check we use on the "frontend" as well (via HTML's `pattern` attribute)

**Review by calpaterson — COMMENTED** [review]
Thank you so much for taking the time to sort this out.  Let me know what you think on the above and then will merge + release

File: csvbase/models.py (hunk 1 above)
Comment (equirke) [reply csvbase/models.py]: Done thanks. I have it checking for the same create_user and update_user in svc.py as well.

**Comment by calpaterson** [comment]
Thanks so much for this.

**Pull request merged**